### ISSUE TITLE
Atom doesn't register as a `public.plain-text` editor on OS X

### ISSUE THREAD

**Issue by Calrion** [atom/atom#11711]
### Prerequisites
- [x] Can you reproduce the problem in [safe mode](http://flight-manual.atom.io/hacking-atom/sections/debugging/#check-if-the-problem-shows-up-in-safe-mode)?

> I haven't actually tested safe mode, because I can't see how it'd be relevant (the problem is determined by Atom, but actually happens outside Atom).
- [X] Are you running the [latest version of Atom](http://flight-manual.atom.io/hacking-atom/sections/debugging/#update-to-the-latest-version)?

> I'm running v1.7.3, which the check updates function reports is the current-latest version.
- [X] Did you check the [debugging guide](http://flight-manual.atom.io/hacking-atom/sections/debugging/)?

> I didn't check the debugging guide because this is about a missing `Info.plist` entry and I can't see how debugging would help.
- [X] Did you check the [FAQs on Discuss](https://discuss.atom.io/c/faq)?

> I found [getting Atom to handle different extensions](https://discuss.atom.io/t/recognizing-different-file-extensions-zmc/27428) and [making Atom recognise a given extension as a particular language](https://discuss.atom.io/t/how-do-i-make-atom-recognize-a-file-with-extension-x-as-language-y/26539) but I don't believe this is an instance of either of those.
- [X] Are you reporting to the [correct repository](https://github.com/atom/atom/blob/master/CONTRIBUTING.md#atom-and-packages)?

> Pretty sure this is the only repository that can control Atom's `Info.plist` file.
- [X] Did you [perform a cursory search](https://github.com/issues?q=is%3Aissue+user%3Aatom+-repo%3Aatom%2Felectron) to see if your bug or enhancement is already reported?

> I searched both open and closed issues for `UTI`, `public.plain-text`, and `"default text editor"` but nothing seemed relevant.

Happy to be corrected about any of this.
### Description

After installing Atom on my Mac, many files (especially those without extensions) still open in (shudder) TextEdit. I took a look in Atom's `Info.plist` file, and also used [duti](http://duti.org/duti.1.html) to check the list of handlers, and it appears that Atom doesn't register for the `public.plain-text` UTI. As a text editor, it really should register with OS X to handle the editing of text files.
### Steps to Reproduce
1. Install Atom v1.7.3 on OS X
2. Open a Terminal window
3. Type `touch MyAtomTestFile` and press return
4. Type `open MyAtomTestFile` and press return

**Expected behavior:** I expected the file to open in Atom

**Actual behavior:** The file opens in TextEdit (or perhaps another editor depending on what you have installed).

(You'll probably also want to `rm MyAtomTestFile`, too.)
### Versions

OS X 10.11.4 (15E65).

`$`**`atom --version`**

```
Atom    : 1.7.3
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1
```

`$`**`apm --version`**

```
apm  1.9.2
npm  2.13.3
node 0.10.40
python 2.7.10
git 2.7.4
```

**Comment by zeckalpha** [comment]
Is this Atom's responsibility? `open -a Atom MyAtomTestFile` works. Or in Finder, Right click the file, Open with... > Atom. Or, to set it globally, select the file in Finder, Command-i, change the Open with: drop down to Atom, click the "change all" button (which should register the handler you describe).

**Comment by pietbrauer** [comment]
@Calrion I think Atom does the right thing. Here is what I've found out:

You create an empty file and get it's `UTI Type`:

```bash
$ file testFile
$ mdls testFile
...
kMDItemContentType             = "public.data"
kMDItemContentTypeTree         = (
    "public.data",
    "public.item"
)
...
```

This gives you the `UTIType` of `public.data`, not `public.plain-text` (as it is an empty file). In Atoms [`Info.plist`](https://github.com/atom/atom/blob/master/resources/mac/atom-Info.plist#L1534-L1551) it registers correctly as an application that can open files of type `public.data`.

The core problem is that you want Atom to be the default application for `public.data`-files. This can only be achieved by setting the `LSHandlerRank` to `Owner` which would be wrong. ([From the `macOS` documentation](https://developer.apple.com/library/ios/documentation/FileManagement/Conceptual/DocumentInteraction_TopicsForIOS/Articles/RegisteringtheFileTypesYourAppSupports.html):

> LSHandlerRank describes whether this application owns the document type or is merely able to open it.

This is also the reason why you see `Atom` listed in the alternative applications in the file menu:
<img width="260" alt="screen shot 2016-08-27 at 14 49 31" src="https://cloud.githubusercontent.com/assets/759730/18027504/b7fc5abe-6c65-11e6-91e9-661a26d97b0d.png">

So how does `TextEdit` become the default for the file type? I did some digging and found that they set a different dictionary for their `public.data` support:

```plist
<dict>
			<key>CFBundleTypeName</key>
			<string>Unknown document</string>
			<key>CFBundleTypeRole</key>
			<string>Viewer</string>
			<key>LSIsAppleDefaultForType</key>
			<true/>
			<key>LSItemContentTypes</key>
			<array>
				<string>public.data</string>
			</array>
			<key>NSDocumentClass</key>
			<string>Document</string>
			<key>NSUbiquitousDocumentUserActivityType</key>
			<string>com.apple.TextEdit.editing</string>
		</dict>
```

__Note the `LSIsAppleDefaultForType`.__

If Atom sets `LSIsAppleDefaultForType=true` in the `Info.plist` as well, it is still not the default application for `public.data` as `TextEdit.app` is still installed.

<img width="260" alt="screen shot 2016-08-27 at 15 07 56" src="https://cloud.githubusercontent.com/assets/759730/18027580/1172cc34-6c68-11e6-964e-d7dcfc5844e0.png">

This bundle key is undocumented, too and I could not find any usages of it outside of Apple applications.

So even if we would add the key to the `Info.plist` the issue would still not be fixed as Apple called dibs first on `public.data`. 😁
